### DISCUSSION TITLE
HASP won't stay connected

### DISCUSSION THREAD

**Discussion by eboston** [HASwitchPlate/HASPone#12] [Q&A]
Things are just getting worse with this HASP.  It won't stay online long enough to do anything with it.  I have another HASP next to it (v0.40) that is having no problems staying online.  It just keeps cycling over and over.

`UPDATE: Checking update URL: https://raw.githubusercontent.com/HASwitchPlate/HASPone/main/update/version.json
[+63.692s] UPDATE: updateEspAvailableVersion: 1.00
[+63.698s] UPDATE: updateLcdAvailableVersion: 3
[+63.702s] UPDATE: Update check completed
[+63.709s] MQTT OUT: 'hasp/desktop/sensor' : '{"espVersion":1.00,"updateEspAvailable":false,"lcdConnected":true,"lcdVersion":"3","updateLcdAvailable":false,"espUptime":63,"signalStrength":-75,"haspName":"desktop","haspIP":"192.168.1.85","haspClientID":"desktop-a4cf12d5ca9d","haspMac":"a4:cf:12:d5:ca:9d","haspManufacturer":"HASwitchPlate","haspModel":"HASPone v1.0.0","heapFree":23728,"heapFragmentation":19,"heapMaxFreeBlockSize":18792,"espCore":"2_7_4"}'
[+63.782s] MQTT: not connected, connecting.
[+63.786s] MQTT: Configuring MQTT connection without TLS.
[+63.793s] HMI OUT: page 0
[+63.798s] HMI OUT: 'p[0].b[1].font=6'
[+63.978s] HMI IN: [5 bytes]: 0x66 0x00 0xff 0xff 0xff
[+63.984s] HMI IN: [sendme Page] '0'
[+63.997s] HMI OUT: 'p[0].b[1].txt="WiFi Connected!\rë Oxford\rIP: 192.168.1.85\r\rMQTT Connecting:\r 192.168.1.200"'
[+64.008s] MQTT: Attempting connection to broker 192.168.1.200 on port 1883 with TLS disabled as clientID desktop-a4cf12d5ca9d
[+64.040s] HMI OUT: 'p[0].b[1].txt="WiFi Connected!\rë Oxford\rIP: 192.168.1.85\r\rMQTT Connected:\r 192.168.1.200"'
[+64.052s] MQTT: connected
[+64.171s] MQTT: subscribed to hasp/desktop/command/#
[+64.180s] MQTT IN: 'hasp/desktop/command/page' : '2'
[+64.186s] HMI OUT: page 2
[+64.276s] MQTT: subscribed to hasp/plates/command/#
[+64.286s] MQTT: subscribed to hasp/desktop/light/switch/#
[+64.352s] MQTT: subscribed to hasp/desktop/brightness/set/#
[+64.484s] MQTT OUT: 'homeassistant/light/desktop/config' : '{"name":"desktop backlight","command_topic":"hasp/desktop/light/switch","state_topic":"hasp/desktop/light/state","brightness_state_topic":"hasp/desktop/brightness/state","brightness_command_topic":"hasp/desktop/brightness/set","availability_topic":"hasp/desktop/status","brightness_scale":100,"unique_id":"desktop-a4cf12d5ca9d-backlight","payload_on":"ON","payload_off":"OFF","payload_available":"ON","payload_not_available":"OFF","device":{"identifiers":["desktop-a4cf12d5ca9d"],"name":"desktop","manufacturer":"HASwitchPlate","model":"HASPone v1.0.0","sw_version":1.00}}'
[+64.666s] MQTT OUT: 'homeassistant/sensor/desktop/config' : '{"name":"desktop sensor","json_attributes_topic":"hasp/desktop/sensor","state_topic":"hasp/desktop/status","unique_id":"desktop-a4cf12d5ca9d-sensor","icon":"mdi:cellphone-text","device":{"identifiers":["desktop-a4cf12d5ca9d"],"name":"desktop","manufacturer":"HASwitchPlate","model":"HASPone v1.0.0","sw_version":1.00}}'
[+64.941s] MQTT OUT: 'homeassistant/number/desktop/config' : '{"name":"desktop active page","command_topic":"hasp/desktop/command/page","state_topic":"hasp/desktop/state/page","retain":true,"optimistic":true,"icon":"mdi:page-next-outline","unique_id":"desktop-a4cf12d5ca9d-page","device":{"identifiers":["desktop-a4cf12d5ca9d"],"name":"desktop","manufacturer":"HASwitchPlate","model":"HASPone v1.0.0","sw_version":1.00}}'
[+65.005s] MQTT OUT: 'hasp/desktop/light/state' : 'ON'
[+65.071s] MQTT OUT: 'hasp/desktop/brightness/state' : 100
[+65.203s] MQTT: subscribed to hasp/desktop/status
[+65.209s] HMI IN: [5 bytes]: 0x66 0x02 0xff 0xff 0xff
[+65.215s] HMI IN: [sendme Page] '2'
[+65.218s] MQTT OUT: 'hasp/desktop/state/json' : '{"event":"page","value":2}'
[+65.226s] MQTT OUT: 'hasp/desktop/stape/page' : '2'
[+83.526s] Reconne▒ting t▒ WiFi networ▒...
[+86.761s] MQTT: not connected, connecting.
[+86.765s] MQTT: Configuring MQTT connection without TLS.
[+86.772s] HMI OUT: page 0
[+86.777s] HMI OUT: 'p[0].b[1].font=6'
[+86.957s] HMI IN: [5 bytes]: 0x66 0x00 0xff 0xff 0xff
[+86.963s] HMI IN: [sendme Page] '0'
[+86.976s] HMI OUT: 'p[0].b[1].txt="WiFi Connected!\rë Oxford\rIP: 192.168.1.85\r\rMQTT Connecting:\r 192.168.1.200"'
[+86.987s] MQTT: Attempting connection to broker 192.168.1.200 on port 1883 with TLS disabled as clientID desktop-a4cf12d5ca9d
[+87.031s] HMI OUT: 'p[0].b[1].txt="WiFi Connected!\rë Oxford\rIP: 192.168.1.85\r\rMQTT Connected:\r 192.168.1.200"'
[+87.042s] MQTT: connected
[+87.055s] MQTT: subscribed to hasp/desktop/command/#
[+87.068s] MQTT IN: 'hasp/desktop/command/page' : '2'
[+87.074s] HMI OUT: page 2
[+87.276s] MQTT: subscribed to hasp/plates/command/#
[+87.357s] MQTT: subscribed to hasp/desktop/light/switch/#
[+87.372s] MQTT: subscribed to hasp/desktop/brightness/set/#
[+87.384s] MQTT OUT: 'homeassistant/light/desktop/config' : '{"name":"desktop backlight","command_topic":"hasp/desktop/light/switch","state_topic":"hasp/desktop/light/state","brightness_state_topic":"hasp/desktop/brightness/state","brightness_command_topic":"hasp/desktop/brightness/set","availability_topic":"hasp/desktop/status","brightness_scale":100,"unique_id":"desktop-a4cf12d5ca9d-backlight","payload_on":"ON","payload_off":"OFF","payload_available":"ON","payload_not_available":"OFF","device":{"▒dentifiers":["desktop-a4cf12d5ca9d"],"name":"desktop","manufacturer":"HASwitchPlate","model":"HASPone v1.0.0","sw_version":1.00}}'
[+87.491s] MQTT OUT: 'homeassistant/sensor/desktop/config' : '{"name":"desktop sensor","json_attributes_topic":"hasp/desktop/sensor","state_topic":"hasp/desktop/status","unique_id":"desktop-a4cf12d5ca9d-sensor","icon":"mdi:cellphone-text","device":{"identifiers":["desktop-a4cf12d5ca9d"],"name":"desktop","manufacturer":"HASwitchPlate","model":"HASPone v1.0.0","sw_version":1.00}}'
[+87.552s] MQTT OUT: 'homeassistant/number/desktop/config' : '{"name":"desktop active page","command_topic":"hasp/desktop/command/page","state_topic":"hasp/desktop/state/page","retain":true,"optimistic":true,"icon":"mdi:page-next-outline","unique_id":"desktop-a4cf12d5ca9d-page","device":{"identifiers":["desktop-a4cf12d5ca9d"],"name":"desktop","manufacturer":"HASwitchPlate","model":"HASPone v1.0.0","sw_version":1.00}}'
[+87.616s] MQTT OUT: 'hasp/desktop/light/state' : 'ON'
[+87.682s] MQTT OUT: 'hasp/desktop/brightness/state' : 100
[+87.807s] MQTT: subscribed to hasp/desktop/status
[+87.814s] HMI IN: [5 bytes]: 0x66 0x02 0xff 0xff 0xff
[+87.819s] HMI IN: [sendme Page] '2'
[+87.822s] MQTT OUT: 'hasp/desktop/state/json' : '{"event":"page","value":2}'
[+87.831s] MQTT OUT: 'hasp/desktop/state/page' : '2'
[+113.276s] Reconnecting to WiFi ▒etwork...
[+116.5s1s] MQTT: not connected, connecting.
[+116.536s] MQTT: Configuring MQTT connection without TLS.
[+116.543s] HMI OUT: page 0
[+116.548s] HMI OUT: 'p[0].b[1].font=6'
[+116.729s] HMI IN: [5 bytes]: 0x66 0x00 0xff 0xff 0xff
[+116.734s] HMI IN: [sendme Page] '0'
[+116.747s] HMI OUT: 'p[0].b[1].txt="WiFi Connected!\rë Oxford\rIP: 192.168.1.85\r\rMQTT Connecting:\r 192.168.1.200"'
[+116.759s] MQTT: Attempting connection to broker 192.168.1.200 on port 1883 with TLS disabled as clientID desktop-a4cf12d5ca9d
[+116.900s] HMI OUT: 'p[0].b[1].txt="WiFi Connected!\rë Oxford\rIP: 192.168.1.85\r\rMQTT Connected:\r 192.168.1.200"'
[+116.912s] MQTT: connected
[+116.922s] MQTT: subscribed to hasp/desktop/command/#
[+116.932s] MQTT IN: 'hasp/desktop/command/page' : '2'
[+116.938s] HMI OUT: page 2
[+117.050s] MQTT: subscribed to hasp/plates/command/#
[+117.059s] MQTT: subscribed to hasp/desktop/light/switch/#
[+117.070s] MQTT: subscribed to hasp/desktop/brightness/set/#
[+117.081s] MQTT OUT: 'homeassistant/light/desktop/config' : '{"name":"desktop backlight","command_topic":"hasp/desktop/light/switch","state_topic":"hasp/desktop/light/state","brightness_state_topic":"hasp/desktop/brightness/state","brightness_command_topic":"hasp/desktop/brightness/set","availability_topic":"hasp/desktop/status","brightness_scale":100,"unique_id":"desktop-a4cf12d5ca9d-backlight","payload_on":"ON","payload_off":"OFF","payload_available":"ON","payload_not_available":"OFF","device":{"identifiers":["desktop-a4cf12d5ca9d"],"name":"desktop","manufacturer":"HASwitchPlate","model":"HASPone v1.0.0","sw_version":1.00}}'
[+117.188s] MQTT OUT: 'homeassistant/sensor/desktop/config' : '{"name":"desktop sensor","json_attributes_topic":"hasp/desktop/sensor","state_topic":"hasp/desktop/status","unique_id":"desktop-a4cf12d5ca9d-sensor","icon":"mdi:cellphone-text","device":{"identifiers":["desktop-a4cf12d5ca9d"],"name":"desktop","manufacturer":"HASwitchPlate","model":"HASPone v1.0.0","sw_version":1.00}}'
[+117.249s] MQTT OUT: 'homeassistant/number/desktop/config' : '{"name":"desktop active page","command_topic":"hasp/desktop/command/page","state_topic":"hasp/desktop/state/page","retain":true,"optimistic":true,"icon":"mdi:page-next-outline","unique_id":"desktop-a4cf12d5ca9d-page","device":{"identifiers":["desktop-a4cf12d5ca9d"],"name":"desktop","manufacturer":"HASwitchPlate","model":"HASPone v1.0.0","sw_version":1.00}}'
[+117.314s] MQTT OUT: 'hasp/desktop/light/state' : 'ON'
[+117.323s] MQTT OUT: 'hasp/desktop/brightness/state' : 100
[+117.335s] MQTT: subscribed to hasp/desktop/status
[+117.342s] HMI IN: [5 bytes]: 0x66 0x02 0xff 0xff 0xff
[+117.347s] HMI IN: [sendme Page] '2'
[+117.351s] MQTT OUT: 'hasp/desktop/state/json' : '{"event":"page","value":2}'
[+117.359s] MQTT OUT: 'hasp/desktop/state/page' : '2'

`

**Comment by ielbury** [comment]
I have problems with my ESP's and DHCP where they sometimes use the wrong IP. I have tracked this down to the fact that I have WiFi repeaters and I have to use static IP's for all ESP nodes. This not only causes the ESP to be flakey but there is some other device in my house that's also flakey.  

To verify, unplug the ESP/plate while pinging the IP.

**Reply by eboston** [reply]
I use static IPs for all my devices so it is not a problem with an IP conflict.

**Reply by ielbury** [reply]
How are you configuring a static IP in HASP? Are you talking about a DHCP IP reservation?

Reservations do not work for me!

**Reply by eboston** [reply]
It is a reservation.  I don't see anywhere in the configuration page (v0.40) to set an static IP address and I can't keep v1.0 connected long enough to get to the configuration page there.

**Reply by ielbury** [reply]
HASP does not have the ability to configure a static IP ... hence why I was asking you how you were doing it.

**Reply by ielbury** [reply]
FYI, This is a repro of the problem I have https://github.com/espressif/esp-idf/issues/2182 ... it happens with Tasmota, WLED, ESPHome + HASP.

It's doing it right now with one my plates (The MAC address changes and multiple IP's are assigned):

![image](https://user-images.githubusercontent.com/240860/110155269-3b676700-7d9a-11eb-8beb-0069f81190a1.png)

This is creates ALL kinds of flakey stuff to happen.

**Reply by ielbury** [reply]
Note that I do not see the problem with ESP32 chips!

**Reply by ielbury** [reply]
Curious, do you have repeaters in your WiFi network? In mine, I have an AP + 2 repeaters.

I was curious about what causes the problem and the solution appears to be to set the CONFIG_LWIP_DHCP_DOES_ARP_CHECK=n in the IDF configuration. It looks like the Arduino IDF configuration does not have this set.

Not my wheel house and certainly not going to solve your problem but it would be nice to know what network configurations cause the problem.

**Reply by eboston** [reply]
> HASP does not have the ability to configure a static IP ... hence why I was asking you how you were doing it.

My mistake.  I misread your question.

**Comment by eboston** [comment]
Flashed back to 0.40 and it seems happy.  Stays connected and I see the time updating.  Flash to 1.0 and the connection problem returns.

**Comment by aderusha** [comment]
Man this is annoying and I want to get to the bottom of this Ed.  There's a lot of the WiFi operation that is something of a black box, specifically the parts that are published by Espressif.  [See here](https://github.com/esp8266/Arduino/issues/7421) for a discussion on that.  That black box includes a lot of details around WiFi timing, reconnects, etc.  I don't know what's different with each release, because Espressif doesn't really say, but the general internet consensus is that `PIO_FRAMEWORK_ARDUINO_ESPRESSIF_SDK22x_190703` is the most compatible.  So, that's what this release is using.

[Here is a list of available options.](https://docs.platformio.org/en/latest/platforms/espressif8266.html#sdk-version)

I've push a build using the more-recent SDK version 191122.  Can you try loading your HASP with the following firmware and see if it behaves any different?

```
https://raw.githubusercontent.com/HASwitchPlate/HASPone/main/Arduino_Sketch/debug/HASwitchPlate.ino.d1_mini-191122.bin
```

Also, I believe you have several HASPs.  What is the scope of this issue?  Are you seeing it on all of your devices?

**Reply by eboston** [reply]
Annoying is an understatement.  I have 4 HASPs deployed around the house and none of those are having this problem so seems like it would not be my WiFi setup.  I have 2 "dev" units I rotate through for testing the 4" display and 1 desktop unit for setting up v1.0 that all show this problem.  I had the desktop unit on 0.40 over night and it was connected fine for a while and then started going through the reconnect ever minute (+/- a few seconds).  On v1.0 it was always reconnecting.  The D1 minis were not all bought at the same time so I find it hard to believe it was faulty units.  I have several other D1 minis in other projects that don't have this issue.  So I don't see any underlying problem that this might be associated with.  To be honest, I have my suspicions about my network (UniFi).

So while writing this response, I have been watching the debug output of my desktop unit running 0.40 and have seen it stay connected for several minutes, go through a few reconnect cycles, then stay connected again for awhile and do it again.  I'll get that other version loaded here and let you know what happens.

Thanks again for all your hard work on this.  I really hate being such a problem for you.  :(

P.S. For future version releases, would it be possible to include the version number in the file name?

**Reply by aderusha** [reply]
There's an interesting datapoint here (maybe).  I have a friend that is experiencing something like the same problem, and is also using UBNT APs.

I'm hearing you on the versioning situation, I need to come up with some better approach than just major release numbers.  Putting the version in the filename is problematic for the update mechanism, but I really need to have a better way to indicate to the user exactly which version they are pulling.

**Reply by eboston** [reply]
I got the new version on and it is now acting more like the v0.40 version.  Stays connected for awhile then goes through a connection recycling period and then stays connected awhile.

**Reply by eboston** [reply]
Just for the fun of it, I set up my old WiFi router creating a new AP and connected the HASP to that.  Let's see if the connection gets better.

**Reply by ielbury** [reply]
> Just for the fun of it, I set up my old WiFi router creating a new AP and connected the HASP to that. Let's see if the connection gets better.

That's a great idea ... i agree 💯 %

**Reply by eboston** [reply]
Hmmm...  Can't find my earlier response.  I setup my old WiFi router on its own subnet and have the HASP connecting through it instead of the UniFi APs.  I have gone over 1.5 hours with no disconnect so it would appear to be something with the configuration of the APs.

**Reply by eboston** [reply]
great.  After I posted that message, my old reply showed up.  :)

**Reply by aderusha** [reply]
> great. After I posted that message, my old reply showed up. :)

It was probably your wifi :D

**Reply by ielbury** [reply]
An option to be able to configure a static IP would be great! This will solve this problem.

**Reply by ielbury** [reply]
> Hmmm... Can't find my earlier response. I setup my old WiFi router on its own subnet and have the HASP connecting through it instead of the UniFi APs. I have gone over 1.5 hours with no disconnect so it would appear to be something with the configuration of the APs.

How many AP/Repeaters do you have in your network (Not the temporary one).

**Reply by eboston** [reply]
4

**Reply by ielbury** [reply]
I am sure that people with a single router/access point have no issues, it the folks with multiple ap's and repeaters that have a problem. There is a problem with how the underlying IDF framework handles dhcp in this situation.

Countless hours have been spent trying to get my head wrapped around this problem and how to solve it ... currently my solution is static ip's.

**Reply by eboston** [reply]
I use static IPs for everything (IP reservations).  I set my router up as an AP and will see how well the connection is.

**Reply by aderusha** [reply]
I have one U6-LR, one UAP-AC-LR, and two UAP-AC-PROs here, all running w/ 1 gbit copper backhaul (so no repeaters).  At any given point I have no fewer than 8 HASPs powered on, and maybe a dozen other various ESP projects scattered around the house.  None of these are using static IPs, and I’ve not seen the sorts of DHCP issues you’ve experienced.

None of this is to discount your experience above,  but it suggests there is some other factor here.  What are you using for DHCP?  Mine is handed out by a pair of win2019 VMs.

**Reply by eboston** [reply]
I don't think it is a DHCP issue.  But my DHCP is handled by the USG.  Having the one HASP connected to the non-UniFi AP seems to be working.  I have it on its own VLAN.

**Reply by aderusha** [reply]
> I don't think it is a DHCP issue. But my DHCP is handled by the USG. Having the one HASP connected to the non-UniFi AP seems to be working. I have it on its own VLAN.

Ahh sorry I was kinda more pointing that at @ielbury but reading above that screenshot suggests he's also using Windows DHCP.

Still trying to figure out the common factor here!

**Reply by eboston** [reply]
The HASP ran all night without dropping the connection.  It also responds to keypresses the way it should.  So I think it is the UniFi APs that are the problem.  If you would be willing to share your configuration, I can compare it to what I have and see if we can find what works.

**Reply by aderusha** [reply]
Allright, this came from a friend who was experiencing similar problems on UBNT:

> did everything here except couldnt find things in red, also disabled PMF and Data Rate Control

![screenshot](https://i.imgur.com/vrwKoqY.png)

**Reply by eboston** [reply]
That pretty much matches what I have already.  Only difference was channel and power.  Channel was set to automatic and power to custom/18dB.  The Airtime Fairness I found under the Radios settings.

**Reply by tribixbite** [reply]
> That pretty much matches what I have already. Only difference was channel and power. Channel was set to automatic and power to custom/18dB. The Airtime Fairness I found under the Radios settings.

I also disabled "Allow BSS Transition with WNM" as according to Unifi support that can interfere with iot devices.

Does your wifi settings page look like this?
![image](https://user-images.githubusercontent.com/381345/110244093-2b32c180-7f2b-11eb-91ad-dd0ad341687f.png)

Presumably I couldnt find the airtime fairness bc I only have one AP. Radios looks like:
![image](https://user-images.githubusercontent.com/381345/110244141-659c5e80-7f2b-11eb-891c-ebc25ff46ac4.png)

**Reply by eboston** [reply]
I disabled the BSS option and disabled PMF.  I have Enable multicast enhancement (IGMPv3) enabled because that was needed for some VLAN stuff I believe.  Otherwise, everything else is the same.

**Reply by eboston** [reply]
Made those changes and switch back to the UniFi AP and the connection problem returned right away.

**Reply by eboston** [reply]
I'm ready to scrap all this UniFi stuff as it has been nothing but problems with connectivity.  Not just the WiFi, but even direct connects have issues at times.

**Reply by tribixbite** [reply]
yeah i can't say i've been pleased with unifi either lol. after i made those changes, clicked 'reconnect', via unifi clients page, then power cycled my HASP, i havent had connection problems. just FYI

**Comment by hockey6611** [comment]
Have there been any developments on this? I had to pause working on my issues due to time, but I'm back at it now. I originally posted in this [post ](https://github.com/HASwitchPlate/HASPone/discussions/14#discussioncomment-450578)but was directed here. I was able to try the SDK version 191122 firmware, but I seem to still be having the same disconnecting issues.

I have 2 HASPs. The first one, I flashed with the beta firmware to try out blueprints, the second one, I waited until the release of 1.0 firmware. The first HASP was upgraded to 1.0 and doesn't have issues. The second HASP has the issues described here with constant disconnects. I have tried erasing and flashing via usb as well with no luck.

**Reply by hockey6611** [reply]
I use unifi APs, pfsense router, and dhcp reservations.

**Reply by eboston** [reply]
I was going to ask if you used UniFi APs like I did.  Not using them solved the problem for me.  I setup an existing WiFi router I have as an AP and the problem went away.  I have several other projects around here using the D1 Mini, but because they don't display when the connection is lost like the HASP does, they could be having the issue, but I'm not seeing it.  I'm thinking of moving all my ESP based devices to the non-UniFi AP.

**Comment by jrdaneman** [comment]
Also running a Unifi network and was having the same problem, then found the @jpelzer 's other thread and it seemed the issues were manifesting exactly the same (mqtt timed out/disconnected, but screen was still responding to taps, and they seemed to break the web-interface hangs) so I decided to try out the test-firmware over there: https://github.com/HASwitchPlate/HASPone/discussions/43#discussioncomment-500315

`https://raw.githubusercontent.com/HASwitchPlate/HASPone/main/Arduino_Sketch/debug/HASwitchPlate.ino.d1_mini-jpelzer.bin`

Have not had any of the network/MQTT disconnects since, though it's only been a day or so. But considering it had been happening minimum once a day, seems like a promising fix.

**Reply by hockey6611** [reply]
Thanks for commenting this! I'll try this version tonight and report back.

**Reply by hockey6611** [reply]
I tried the jpelzer firmware and I still have the same issues. I don't have a non-Unifi AP to test at the moment, but I will test this next, to see if this works like @eboston

**Comment by hockey6611** [comment]
I continue to have issues with one of my HASPs. One connects to my unifi APs and I have no issues. A second HASP continues to have issues with disconnecting. I have also tried a separate AP it seems to keep a connection a little longer, but ultimately it still disconnects from the network.

Has anyone else in this thread with the disconnecting issue been able to identify any resolutions? I have also flashed the latest firmware (1.02), and still no improvement.

**Reply by shepler007** [reply]
Have you tried separating your 2G and 5G with different SSID's and only use the 2.4 for the HASPs?  I had to do this years ago to make all my ESP's on my network reliable.

**Reply by hockey6611** [reply]
Yep. I have tried to separate the 2.4G and 5G SSIDs on both the Unifi APs and on the non-Unifi AP. I have not had issues with other ESPs on Unifi APs. The one HASP I have working, happily works on the Unifi APs with same 2.4G/5G SSIDs. The second HASP I have trouble with has the same issue on the non-unifi AP with separate APs.

**Reply by aderusha** [reply]
@hockey6611 did you purchase this as an assembled unit or was this a DIY build?

**Reply by hockey6611** [reply]
Both HASPs I have were pre-assembled purchases from your Tindie store.

**Reply by aderusha** [reply]
If you'd like, message me on the Tindie store with your order number and I can replace the hardware.  If you're comfortable disassembling the thing, I can send you the PCB assembly, you can swap that out, and see if it resolves your issue.

**Reply by hockey6611** [reply]
Thank you for the offer of that! I will reach out on the Tindie store

**Reply by hockey6611** [reply]
Thank you @aderusha! I received the replacement HASP a while ago, and finally was able to test it. I can confirm it works well on my Unifi AP, just as my other HASP does. Must have been a hardware issue.

**Comment by wr420** [comment]
I am also experiencing issues similar to what is described here. 
I am on the v1.1 hasp firmware and I recall the v0.40 hasp devices working more reliably and being more responsive than the 1.0 and 1.1 versions.  I have about 6 of them around the house and I see most of them disconnect constantly.  I have 2, that are the original 2 I set up, and upgraded from v0.40, and they seem to be the most consistent(at working).  
One device in particular was giving me a really hard time, button presses did not work, seemed locked up.  I switched the baud back to 115200 and it started working better.  I have not switched it back to confirm it it gets worse again.  The other 5 are still at 921600.

My WIFI network is (cringe) openmesh/cloudtrax, (I would call it day and consider this the issue if I didn't see others with a similar issue;)  I am running separate 2.4 and 5 networks and it does help.  I have also added another access point near buy, and it helped but... the problem is not gone 100%.  My devices are DHCP w reservations so I may go static and try that.

I also have several other ESP related switches and sensors in the area that do not seem to have the disconnection issue.  One in particular is just a gpio line tied to a reed switch that controls a lamp in HA.  The lamp comes on almost instantly every time.

These are DIY units.  Please ask for clarification if any is needed to help track this down.

AND THANK YOU (LUMA) and to everyone who has taken part in this great project!  
love love love it


EDIT: v1.02 has made a huge difference, dare I say... fixed.
Thank You!

**Comment by jphaus01** [comment]
I know this is an old thread, but I'm a new HASP user (bought fully assembled directly from Etsy and I'm having the same disconnecting problem as noted above. Was this found to be a hardware issue?

2025-12-26 16:19:53: Client closed its connection.
2025-12-26 16:20:08: Client plate01-8caab55c6cca has exceeded timeout, disconnecting.
2025-12-26 16:20:08: New connection from 10.10.10.51:62210 on port 1883.
2025-12-26 16:20:08: New client connected from 10.10.10.51:62210 as plate01-8caab55c6cca (p2, c1, k30, u'mqttuser').
2025-12-26 16:20:54: Client plate01-8caab55c6cca has exceeded timeout, disconnecting.
2025-12-26 16:21:09: New connection from 10.10.10.51:50165 on port 1883.
2025-12-26 16:21:09: New client connected from 10.10.10.51:50165 as plate01-8caab55c6cca (p2, c1, k30, u'mqttuser').
2025-12-26 16:22:16: Client plate01-8caab55c6cca has exceeded timeout, disconnecting.
2025-12-26 16:22:16: New connection from 10.10.10.51:57390 on port 1883.
2025-12-26 16:22:16: New client connected from 10.10.10.51:57390 as plate01-8caab55c6cca (p2, c1, k30, u'mqttuser').
2025-12-26 16:23:03: Client plate01-8caab55c6cca has exceeded timeout, disconnecting.
2025-12-26 16:23:04: New connection from 10.10.10.51:55256 on port 1883.
2025-12-26 16:23:04: New client connected from 10.10.10.51:55256 as plate01-8caab55c6cca (p2, c1, k30, u'mqttuser').
2025-12-26 16:23:50: Client plate01-8caab55c6cca has exceeded timeout, disconnecting.
2025-12-26 16:24:05: New connection from 10.10.10.51:52960 on port 1883.
2025-12-26 16:24:05: New client connected from 10.10.10.51:52960 as plate01-8caab55c6cca (p2, c1, k30, u'mqttuser').
2025-12-26 16:24:51: Client plate01-8caab55c6cca has exceeded timeout, disconnecting.
2025-12-26 16:25:06: New connection from 10.10.10.51:51774 on port 1883.
2025-12-26 16:25:06: New client connected from 10.10.10.51:51774 as plate01-8caab55c6cca (p2, c1, k30, u'mqttuser').
2025-12-26 16:25:59: Client plate01-8caab55c6cca has exceeded timeout, disconnecting.

**Reply by aderusha** [reply]
It can be difficult to troubleshoot wifi problems from the other side of this connection, so if that’s actually the problem it might be useful to open the device for serial access.  Annoyingly this means removing it from the wall.  A set of four philips #1 screws hold the rear enclosure on, inside there are a pair of 1.5mm hex head screws holding the PCB in.  Pulling that out gives you access to a USB port that can power the device (make sure its not connected to AC), and can give you detailed logging to better see what the heck is going on.
